test/integration/controller_test.go

```diff
@@ -0,0 +1,162 @@
+	"k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/kubernetes/pkg/util/wait"
+
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/v1alpha1"
+	"github.com/kubernetes-incubator/service-catalog/pkg/brokerapi"
+	fakebrokerapi "github.com/kubernetes-incubator/service-catalog/pkg/brokerapi/fake"
+	"github.com/kubernetes-incubator/service-catalog/pkg/client/clientset_generated/clientset"
+	scinformers "github.com/kubernetes-incubator/service-catalog/pkg/client/informers"
+	informers "github.com/kubernetes-incubator/service-catalog/pkg/client/informers/servicecatalog/v1alpha1"
+	"github.com/kubernetes-incubator/service-catalog/pkg/controller/wip"
+	"github.com/kubernetes-incubator/service-catalog/pkg/registry/servicecatalog/server"
+)
+
+// TestController is a very basic test to start
+//
+// need etcd running
+// start a fresh apiserver for the controller to talk to
+func TestController(t *testing.T) {
+	fakeKubeClient, catalogClient, fakeBrokerCatalog, _, _, testController, _, stopCh := newTestController(t)
+	defer close(stopCh)
+
+	t.Log(fakeKubeClient, catalogClient, fakeBrokerCatalog, testController, stopCh)
+
+	fakeBrokerCatalog.RetCatalog = &brokerapi.Catalog{
+		Services: []*brokerapi.Service{
+			{
+				Name:        "test-service",
+				ID:          "12345",
+				Description: "a test service",
+				Plans: []brokerapi.ServicePlan{
+					{
+						Name:        "test-plan",
+						Free:        true,
+						ID:          "34567",
+						Description: "a test plan",
+					},
+				},
+			},
+		},
+	}
+	name := "test-name"
+	broker := &v1alpha1.Broker{
+		ObjectMeta: v1.ObjectMeta{Name: name},
+		Spec: v1alpha1.BrokerSpec{
+			URL:     "https://example.com",
+			OSBGUID: "OSBGUID field",
+		},
+	}
+	brokerClient := catalogClient.Servicecatalog().Brokers()
+
+	brokerServer, err := brokerClient.Create(broker)
+	if nil != err {
+		t.Fatalf("error creating the broker %q (%q)", broker, err)
+	}
+
+	if err := wait.PollImmediate(500*time.Millisecond, wait.ForeverTestTimeout,
+		func() (bool, error) {
+			brokerServer, err = brokerClient.Get(name)
+			if nil != err {
+				return false,
+					fmt.Errorf("error getting broker %s (%s)",
+						name, err)
+			} else if len(brokerServer.Status.Conditions) > 0 {
+				t.Log(brokerServer)
+				return true, nil
+			} else {
+				return false, nil
+			}
+		},
+	); err != nil {
+		t.Fatal(err)
+	}
+
+	// check
+	serviceClassClient := catalogClient.Servicecatalog().ServiceClasses()
+	_, err = serviceClassClient.Get("test-service")
+	if nil != err {
+		t.Fatal("could not find the test service", err)
+	}
+
+	// cleanup our broker
+	err = brokerClient.Delete(name, &v1.DeleteOptions{})
+	if nil != err {
+		t.Fatalf("broker should be deleted (%s)", err)
+	}
+
+	// uncomment if/when deleting a broker deletes the associated service
+	// if class, err := serviceClassClient.Get("test-service"); nil == err {
+	// 	t.Fatal("found the test service that should have been deleted", err, class)
+	// }
```

Comment:
@pmorie any idea why this clause doesn't seem to work? I did put in a sleep and a retry loop, but no final effect.